src/services/cryptography/crypto.ts

```diff
@@ -1,46 +1,65 @@
-} from '@govtechsg/oa-encryption';
-import { computeEntryHash } from '@veramo/utils';
-import { ICryptographyService } from './index';
+import crypto from 'crypto';
+import { HashAlgorithm, EncryptionAlgorithm, ICryptographyService, ivLengthMap, keyLengthMap } from './index';
 
 /**
  * Service for cryptography operations.
  */
 export class CryptographyService implements ICryptographyService {
     /**
      * Computes the hash of the input string.
-     * @param input - The input string to compute the hash for.
-     * @returns The computed hash.
+     * @param {string} input - The input string to compute the hash for.
+     * @param {HashAlgorithm} [algorithm=HashAlgorithm.SHA_256] - The hash algorithm to use.
+     * @returns {string} The computed hash as a hexadecimal string.
      */
-    computeHash(input: string) {
-        const hash = computeEntryHash(input);
+    computeHash(input: string, algorithm: HashAlgorithm = HashAlgorithm.SHA_256) {
+        const hash = crypto.createHash(algorithm).update(input).digest('hex');
 
         return hash;
     }
 
     /**
-     * Generates an encryption key.
-     * @returns The generated encryption key.
+     * Generates an encryption key for the specified algorithm.
+     * @param {EncryptionAlgorithm} [algorithm=EncryptionAlgorithm.AES_256_GCM] - The encryption algorithm to generate the key for.
+     * @returns {string} The generated encryption key as a hexadecimal string.
      */
-    generateEncryptionKey() {
-        const encryptionKey = oaGenerateEncryptionKey();
+    generateEncryptionKey(algorithm: EncryptionAlgorithm = EncryptionAlgorithm.AES_256_GCM) {
+        const encryptionKey = crypto.randomBytes(keyLengthMap[algorithm]).toString('hex');
 
         return encryptionKey;
     }
 
     /**
-     * Encrypts a string using the provided key.
-     * @param input - The string to encrypt.
-     * @param key - The encryption key to use.
-     * @returns The encrypted document.
+     * Encrypts a string using the provided key with the specified encryption algorithm.
+     * @param {string} input - The string to encrypt.
+     * @param {string} key - The encryption key to use (hexadecimal string).
+     * @param {EncryptionAlgorithm} [algorithm=EncryptionAlgorithm.AES_256_GCM] - The encryption algorithm to use.
+     * @returns {IEncryptionResult} An object containing the encrypted data:
+     *          - cipherText: The encrypted string (Base64 encoded).
+     *          - iv: The initialization vector used for encryption (Base64 encoded).
+     *          - tag: The authentication tag for GCM mode (Base64 encoded).
+     *          - type: The encryption algorithm used.
+     * @throws {Error} If the key length is invalid for the specified algorithm.
      */
-    encryptString(input: string, key: string) {
-        const encryptedDocument = oaEncryptString(input, key);
+    encryptString(input: string, key: string, algorithm = EncryptionAlgorithm.AES_256_GCM) {
```

Comment:
I just curious about this function. Why don't we use the library like `@govtechsg/oa-encryption` for encryption?

Reply:
@ldhyen99, this is a good question. 

The main reason is to decouple the encryption method from the way in which Open Attestation does it.

For example, in the current implementation, `@govtechsg/oa-encryption` uses an OPEN-ATTESTATION-TYPE-1. Also, what happens if OA no longer wants to use the AES-GMC algorithm?